front/src/containers/Islands/WfIslandAggChild.tsx

```diff
@@ -1,42 +1,71 @@
-import React, { useState, useEffect, useRef } from 'react';
-import SortKind from 'containers/AggDropDown/SortKind';
-import CustomDropDown from 'containers/AggDropDown/CustomDrop';
+import * as FontAwesome from 'react-fontawesome';
+
+import { FormControl, Panel } from 'react-bootstrap';
+// import aggToField from 'utils/aggs/aggToField';
+// import { capitalize } from 'utils/helpers';
+import {
+  PresearchTitle,
+  ThemedPresearchHeader,
+} from 'components/StyledComponents';
+import React, { useEffect, useRef, useState } from 'react';
+import RichTextEditor, { EditorValue } from 'react-rte';
+import { deleteCrowdKeyValueId, insertCrowdKeyValueId, updateCrowdKeyValueId } from '../../services/crowdKeys/actions'
+import { deleteLabelMutation, fetchSuggestedLabels, setShowLoginModal, upsertLabelMutation } from '../../services/study/actions'
 import { useDispatch, useSelector } from 'react-redux';
-import { RootState } from 'reducers';
+
 import { BeatLoader } from 'react-spinners';
-import { fetchSuggestedLabels, upsertLabelMutation, deleteLabelMutation, setShowLoginModal } from '../../services/study/actions'
-import { insertCrowdKeyValueId, deleteCrowdKeyValueId } from '../../services/crowdKeys/actions'
+import CustomDropDown from 'containers/AggDropDown/CustomDrop';
+import { RootState } from 'reducers';
+import SortKind from 'containers/AggDropDown/SortKind';
+import ThemedButton from 'components/StyledComponents/index';
 
 interface Props {
   aggId?: string;
   nctId: string;
+  suggestedLabels: any;
 }
 
 function WfIslandAggChild(props: Props) {
 
-  const { aggId, nctId } = props;
+  const { aggId, nctId, suggestedLabels } = props;
   const dispatch = useDispatch();
   const emptySet = new Set();
 
   const isLoading = useSelector((state: RootState) => state.study.isFetchingSuggestedLables);
-  const suggestedLabels = useSelector((state: RootState) => state.study.suggestedLabels);
   const islandConfig = useSelector((state: RootState) => state.search.islandConfig);
   const user = useSelector((state: RootState) => state.user.current);
   const isUpsertingLabel = useSelector((state: RootState) => state.study.isUpsertingLabel)
 
-
   let getCurrentAgg = () => {
     let jsonConfig = islandConfig
     return aggId && jsonConfig[aggId]
   }
   let currentAgg = getCurrentAgg();
 
+  const crowdKeyValueData = suggestedLabels?.data?.crowd_key_value_ids.filter(x => x.crowd_key == currentAgg.name) || [];
+  let crowdKeyValueTitle = null;
+
+  if (currentAgg.display === "TEXT_EDITOR" && crowdKeyValueData.length > 0) {
```

Comment:
This page now has 2 supported island types - TEXT_EDITOR and (implied to be CHECKBOX).  Would like to discuss how code is organized for those 2 cases and how to flow more clearly for what is shared behavior vs unique behavior and set-up for support of other types as needed (mainly dropdown, possibly alternative text (React-rte) or others. Possibly comment on sections/properties of the panel and clarify the data source and actions - also possibly reconcile with the search islands agg and crowdagg